relaynet-cogrpc-jvm/src/main/java/tech/relaycorp/relaynet/cogrpc/server/CogRPCService.kt

```diff
@@ -0,0 +1,97 @@
+package tech.relaycorp.relaynet.cogrpc.server
+
+import com.google.protobuf.ByteString
+import io.grpc.stub.StreamObserver
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import tech.relaycorp.relaynet.CargoRelay
+import tech.relaycorp.relaynet.CargoRelayServer
+import tech.relaycorp.relaynet.cogrpc.CargoDelivery
+import tech.relaycorp.relaynet.cogrpc.CargoDeliveryAck
+import tech.relaycorp.relaynet.cogrpc.CargoRelayGrpc
+import java.util.logging.Level
+import java.util.logging.Logger
+
+class CogRPCService(
+    private val coroutineScope: CoroutineScope,
+    private val connectionService: CargoRelayServer.ConnectionService
+) : CargoRelayGrpc.CargoRelayImplBase() {
+
+    override fun deliverCargo(responseObserver: StreamObserver<CargoDeliveryAck>) =
+        object : StreamObserver<CargoDelivery> {
+            override fun onNext(cargoDelivery: CargoDelivery) {
+                coroutineScope.launch {
+                    logger.info("deliverCargo next")
+                    connectionService.deliverCargo(cargoDelivery.toMessageReceived())
+                    logger.info("deliverCargo next ack")
```

Comment:
I like the logging, but what do you think if we demote the level to `debug` and include the `cargoDelivery.id` in the message? I suspect that'll come in handy when troubleshooting in the future.

Also, can we enable debug logs during development but skip debug/info logs on end users' devices? It wouldn't be ideal to leave behind an audit trail for privacy/safety reasons.

Reply:
I'll look into setting the log level for this.